packages/happy-dom/src/event/events/CustomEvent.ts

```diff
@@ -5,20 +5,18 @@ import ICustomEventInit from './ICustomEventInit.js';
  *
  */
 export default class CustomEvent extends Event {
-	public detail: object = null;
+	public detail: object;
```

Comment:
```suggestion
	public detail: object | string | number | boolean;
```

Reply:
Should this be `any` like i TypeScript lib.dom.d.ts: https://github.com/microsoft/TypeScript/blob/695226ea5ac64195e4ab0180cd38f06010157ec0/src/lib/dom.generated.d.ts#L5891 and in the spec: https://dom.spec.whatwg.org/#interface-customevent

> readonly attribute [any](https://webidl.spec.whatwg.org/#idl-any) [detail](https://dom.spec.whatwg.org/#dom-customevent-detail);

Reply:
Yes, I guess that "any" is most correct. Not sure that the current TS config allows it, but maybe you can try to override it otherwise.

I'm thinking about going through the types in general in Happy DOM and make sure they follow the lib.dom.d.ts.